src/lib.rs

```diff
@@ -551,10 +550,15 @@ pub mod contract {
             let calculate_swap_result =
                 self.calculate_swap(pool_key, x_to_y, amount, by_amount_in, sqrt_price_limit)?;
 
+            let mut crossed_tick_indexes: Vec<i32> = vec![];
+
             for tick in calculate_swap_result.ticks.iter() {
                 self.ticks.update(pool_key, tick.index, tick)?;
+                crossed_tick_indexes.push(tick.index);
             }
 
+            self.emit_cross_tick_event(caller, pool_key, crossed_tick_indexes);
```

Comment:
do not emit event in case there was no cross. statistically >95% of swaps have no cross